services/resource-usage-tracker/src/simcore_service_resource_usage_tracker/modules/db/repositories/resource_tracker.py

```diff
@@ -267,3 +299,266 @@ async def total_service_runs_by_product_and_wallet(
 
         row = result.first()
         return cast(PositiveInt, row[0]) if row else 0
+
+    #################################
+    # Credit transactions
+    #################################
+
+    async def create_credit_transaction(
+        self, data: CreditTransactionCreate
+    ) -> CreditTransactionId:
+        async with self.db_engine.begin() as conn:
+            insert_stmt = (
+                resource_tracker_credit_transactions.insert()
+                .values(
+                    product_name=data.product_name,
+                    wallet_id=data.wallet_id,
+                    wallet_name=data.wallet_name,
+                    pricing_plan_id=data.pricing_plan_id,
+                    pricing_detail_id=data.pricing_detail_id,
+                    user_id=data.user_id,
+                    user_email=data.user_email,
+                    credits=data.credits,
+                    transaction_status=data.transaction_status,
+                    transaction_classification=data.transaction_classification,
+                    service_run_id=data.service_run_id,
+                    payment_transaction_id=data.payment_transaction_id,
+                    created=data.created_at,
+                    last_heartbeat_at=data.last_heartbeat_at,
+                    modified=sa.func.now(),
+                )
+                .returning(resource_tracker_credit_transactions.c.transaction_id)
```

Comment:
in order to make codeclimate much happier I think you could change these huge calls into this:
```.values(jsonable_encoder(data, include={"product_name", "wallet_id", ...})``` or all of them if you use all of them. then code climate happy, everyone happy ;)